sdk/lib/opentelemetry/sdk/resources/detectors/google_cloud_platform.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'google-cloud-env'
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      module Detectors
+        # GoogleCloudPlatform contains detect class method for determining gcp environment resource labels
+        module GoogleCloudPlatform
+          extend self
+
+          def detect
+            gcp_env = Google::Cloud::Env.new
+            resource_labels = {}
+
+            if gcp_env.compute_engine?
+              resource_labels[CLOUD_RESOURCE[:provider]] = 'gcp'
+              resource_labels[CLOUD_RESOURCE[:account_id]] = gcp_env.project_id || ''
```

Comment:
do we want blank strings if it's not there?  we'll end up increasing the payload by sizeof(key)

if you're wondering why they wouldn't be there, https://github.com/GoogleCloudPlatform/k8s-metadata-proxy is a thing.

Reply:
I have a feeling workload concealment does something similar as well.

Reply:
Yeah, I prefer to not have the attribute rather than have an empty attribute.

Reply:
Updated to remove keys with blank values.
https://github.com/open-telemetry/opentelemetry-ruby/pull/263/files#diff-f799130e405e5a80aa43ba547be52cdbR40

https://github.com/open-telemetry/opentelemetry-ruby/pull/263/files#diff-6e5e6a6b9a0554f85023aa90b68f94abR62-R76